Fight.java

```diff
@@ -1,43 +1,143 @@
-        if(stamina1 <= 0) {
-          fighting = false;
-          return 2;
-        }
-        // Hawk attacks.
-        int random2 = (int)(Math.random() * two.speed());
-        if(random2 > 0) {
-          stamina2 -= one.attack() * two.defense();
-          System.out.println("Stamina 2: " + stamina2);
-        }
-        if(stamina2 <= 0) {
-          fighting = false;
-          return 1;
-        }
+  public static void duel(IDuelable one, IDuelable two, Environment currentEnvironment) {
+    System.out.println(currentEnvironment.environment());
+    JPanel game = new JPanel();
+    game.setLayout(null);
+    frame = new JFrame();
+    frame.setSize(700, 300);
+    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+    frame.add(game);
+
+    animalHealth1 = new JProgressBar(0, 10000);
+    animalHealth1.setForeground(Color.red);
+    animalHealth1.setBackground(Color.darkGray);
+    animalHealth1.setBounds(10, 10, 500, 40);
+    animalHealth1.setValue(10000);
+    animalHealth1.setStringPainted(true);
+
+    JLabel animalOne = new JLabel(one.name().toUpperCase());
+    animalOne.setBounds(515,10,85,40);
+    animalOne.setFont(new Font(null, Font.BOLD, 20));
+
+    animalHealth2 = new JProgressBar(0, 10000);
+    animalHealth2.setForeground(Color.red);
+    animalHealth2.setBackground(Color.darkGray);
+    animalHealth2.setBounds(10,100,500,40);
+    animalHealth2.setValue(10000);
+    animalHealth2.setStringPainted(true);
+
+    JLabel animalTwo = new JLabel(two.name().toUpperCase());
+    animalTwo.setBounds(515,100,85,40);
+    animalTwo.setFont(new Font(null, Font.BOLD, 20));
+
+    JLabel winner = new JLabel();
+    winner.setBounds(10,200,400,100);
+    winner.setFont(new Font(null, Font.BOLD, 30));
+
+    backButton = new JButton("Back");
+    backButton.setBounds(595, 205, 95, 25);
+    backButton.addActionListener(new Fight());
+
+    exitButton = new JButton("Exit");
+    exitButton.setBounds(595, 235, 95, 25);
+    exitButton.addActionListener(new Fight());
+
+    rematchButton = new JButton("Rematch");
+    rematchButton.setBounds(595, 175, 95, 25);
+    rematchButton.addActionListener(new Fight());
+
+
+    game.add(animalHealth1);
+    game.add(animalOne);
+    game.add(animalHealth2);
+    game.add(animalTwo);
+    game.add(winner);
+    game.add(backButton);
+    game.add(exitButton);
+    game.add(rematchButton);
+
+    frame.setVisible(true);
+    int delay = 100;
+    ActionListener taskPerformer = new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
+        int health1 = animalHealth1.getValue();
+        int health2 = animalHealth2.getValue();
+        if (animalHealth1.getValue() > 0 && animalHealth2.getValue() > 0) {
+          health1 -= attack1(one, two, currentEnvironment);
+          animalHealth1.setValue(health1);
+          animalHealth1.setString(health1+"/"+10000+" HP");
+          if(animalHealth1.getValue() > 0) {
+            health2 -= attack2(one, two, currentEnvironment);
```

Comment:
Notice that `attact2` is basically the same as `attact1` except with the dualables reversed; you could just get rid of `attack2` and just reuse `attact1`:
```suggestion
            health2 -= attack1(two, one, currentEnvironment);
```